### PR TITLE
TIP-841: Move the channel classes to Akeneo\Channel 

### PR DESCRIPTION
**Pull request by arnolanglade**
**Description (for Contributor and Core Developer)**

JJ made this first PR https://github.com/akeneo/pim-community-dev/pull/8075 to move locale to a new bounded context called `Channel`. This PR moves all classes related to the channel to this new context.

```
├── Bundle
│   ├── AkeneoChannelBundle.php
│   ├── Controller
│   │   ├── ExternalApi
│   │   │   ├── ChannelController.php
│   │   │   └── LocaleController.php
│   │   └── InternalApi
│   │       ├── ChannelController.php
│   │       └── LocaleController.php
│   ├── DependencyInjection
│   │   ├── AkeneoChannelExtension.php
│   │   └── CompilerPass
│   │       └── ResolveDoctrineTargetModelPass.php
│   ├── Doctrine
│   │   ├── Remover
│   │   │   └── ChannelRemover.php
│   │   └── Repository
│   │       ├── ChannelRepository.php
│   │       └── LocaleRepository.php
│   ├── EventListener
│   │   └── ChannelLocaleSubscriber.php
│   ├── Resources
│   │   └── config
│   │       ├── controllers.yml
│   │       ├── doctrine
│   │       │   └── model
│   │       │       ├── Channel.orm.yml
│   │       │       ├── ChannelTranslation.orm.yml
│   │       │       └── Locale.orm.yml
│   │       ├── jobs.yml
│   │       ├── processors.yml
│   │       ├── readers.yml
│   │       ├── routing
│   │       │   ├── external_api
│   │       │   │   ├── channel.yml
│   │       │   │   └── locale.yml
│   │       │   └── internal_api
│   │       │       ├── channel.yml
│   │       │       └── locale.yml
│   │       ├── routing.yml
│   │       ├── services.yml
│   │       ├── steps.yml
│   │       ├── validation
│   │       │   └── channel.yml
│   │       └── writers.yml
│   └── spec
│       ├── Doctrine
│       │   ├── Remover
│       │   │   └── ChannelRemoverSpec.php
│       │   └── Repository
│       │       ├── ChannelRepositorySpec.php
│       │       └── LocaleRepositorySpec.php
│       └── EventListener
│           └── ChannelLocaleSubscriberSpec.php
└── Component
    ├── Model
    │   ├── ChannelInterface.php
    │   ├── Channel.php
    │   ├── ChannelTranslationInterface.php
    │   ├── ChannelTranslation.php
    │   ├── LocaleInterface.php
    │   └── Locale.php
    ├── Normalizer
    │   ├── ExternalApi
    │   │   ├── ChannelNormalizer.php
    │   │   └── LocaleNormalizer.php
    │   ├── InternalApi
    │   │   ├── ChannelNormalizer.php
    │   │   └── LocaleNormalizer.php
    │   ├── Standard
    │   │   ├── ChannelNormalizer.php
    │   │   └── LocaleNormalizer.php
    │   └── Versioning
    │       ├── ChannelNormalizer.php
    │       └── LocaleNormalizer.php
    ├── Repository
    │   ├── ChannelRepositoryInterface.php
    │   └── LocaleRepositoryInterface.php
    ├── spec
    │   ├── Normalizer
    │   │   ├── ExternalApi
    │   │   │   ├── ChannelNormalizerSpec.php
    │   │   │   └── LocaleNormalizerSpec.php
    │   │   ├── InternalApi
    │   │   │   ├── ChannelNormalizerSpec.php
    │   │   │   └── LocaleNormalizerSpec.php
    │   │   ├── Standard
    │   │   │   ├── ChannelNormalizerSpec.php
    │   │   │   └── LocaleNormalizerSpec.php
    │   │   └── Versioning
    │   │       ├── ChannelNormalizerSpec.php
    │   │       └── LocaleNormalizerSpec.php
    │   ├── Updater
    │   │   ├── ChannelUpdaterSpec.php
    │   │   └── LocaleUpdaterSpec.php
    │   └── Validator
    │       └── Constraint
    │           ├── LocaleSpec.php
    │           └── LocaleValidatorSpec.php
    ├── tests
    │   └── integration
    │       └── Channel
    │           ├── StandardNormalizationIntegration.php
    │           └── UpdaterIntegration.php
    ├── Updater
    │   ├── ChannelUpdater.php
    │   └── LocaleUpdater.php
    └── Validator
        └── Constraint
            ├── ActivatedLocale.php
            ├── ActivatedLocaleValidator.php
            ├── Locale.php
            └── LocaleValidator.php

46 directories, 67 files
```

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | yes
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -61,3 +61,19 @@ find ./src/ -type f -print0 | xargs -0 sed -i 's/Akeneo\\Bundle\\FileStorageBund
 find ./src/ -type f -print0 | xargs -0 sed -i 's/Akeneo\\Component\\Classification/Akeneo\\Tool\\Component\\Classification/g'
 find ./src/ -type f -print0 | xargs -0 sed -i 's/Akeneo\\Bundle\\ClassificationBundle/Akeneo\\Tool\\Bundle\\ClassificationBundle/g'
 find ./src/ -type f -print0 | xargs -0 sed -i 's/Akeneo\\Bundle\\BufferBundle/Akeneo\\Tool\\Bundle\\BufferBundle/g'
+find ./src/ -type f -print0 | xargs -0 sed -i 's/Akeneo\\Tool\\Bundle\\ApiBundle\\Controller\\ChannelController/Akeneo\\Channel\\Bundle\\Controller\\ExternalApi\\ChannelController/g'
+find ./src/ -type f -print0 | xargs -0 sed -i 's/Akeneo\\Tool\\Bundle\\ApiBundle\\Controller\\ChannelController/Akeneo\\Channel\\Bundle\\Controller\\ExternalApi\\ChannelController/g'
```
Comment (pierallard) [inline UPGRADE.md]: This is the same line than above

File: UPGRADE.md [hunk 2]
```diff
@@ -61,3 +61,19 @@ find ./src/ -type f -print0 | xargs -0 sed -i 's/Akeneo\\Bundle\\FileStorageBund
...
+find ./src/ -type f -print0 | xargs -0 sed -i 's/Pim\\Bundle\\EnrichBundle\\Normalizer\\ChannelNormalizer/Akeneo\\Channel\\Component\\Normalizer\\InternalApi\\ChannelNormalizer/g'
+find ./src/ -type f -print0 | xargs -0 sed -i 's/Pim\\Component\\Catalog\\Normalizer\\Standard\\ChannelNormalizer/Akeneo\\Channel\\Component\\Normalizer\\Standard\\ChannelNormalizer/g'
+find ./src/ -type f -print0 | xargs -0 sed -i 's/Pim\\Bundle\\VersioningBundle\\Normalizer\\Flat\\ChannelNormalizer/Akeneo\\Channel\\Component\\Normalizer\\Versioning\\ChannelNormalizer/g'
+find ./src/ -type f -print0 | xargs -0 sed -i 's/Pim\\Component\\Catalog\\Repository\\ChannelRepositoryInterface/Akeneo\\Channel\\Component\\Repository\\ChannelRepositoryInterface/g'
+find ./src/ -type f -print0 | xargs -0 sed -i 's/Pim\\Component\\Catalog\\Updater\\ChannelUpdater/Akeneo\\Channel\\Component\\Updater\\ChannelUpdater/g'
+find ./src/ -type f -print0 | xargs -0 sed -i 's/Pim\\Component\\Catalog\\Updater\\LocalelUpdater/Akeneo\\Channel\\Component\\Updater\\LocalelUpdater/g'
```
Comment (pierallard) [inline UPGRADE.md]: LocalelUpdater ?

File: src/Akeneo/Channel/Bundle/Controller/ExternalApi/ChannelController.php [hunk 3]
```diff
@@ -304,9 +304,9 @@ protected function getDecodedContent($content)
     /**
      * Update a channel. It throws an error 422 if a problem occurred during the update.
      *
-     * @param ChannelInterface        $channel
-     * @param array                   $data
-     * @param string                  $anchor
+     * @param \Akeneo\Channel\Component\Model\ChannelInterface $channel
```
Comment (pierallard) [inline src/Akeneo/Channel/Bundle/Controller/ExternalApi/ChannelController.php]: use

Comment (pierallard) [reply src/Akeneo/Channel/Bundle/Controller/ExternalApi/ChannelController.php]: (And there is again some under)

File: src/Akeneo/Channel/Bundle/EventListener/ChannelLocaleSubscriber.php [hunk 4]
```diff
@@ -28,34 +27,26 @@ class ChannelLocaleSubscriber implements EventSubscriberInterface
...
         BulkSaverInterface $saver,
-        CompletenessRemoverInterface $completenessRemover,
-        CommandLauncher $commandLauncher = null,
-        TokenStorageInterface $tokenStorage = null
+        CommandLauncher $commandLauncher,
+        TokenStorageInterface $tokenStorage
     ) {
         $this->repository = $repository;
         $this->saver = $saver;
-        $this->completenessRemover = $completenessRemover;
```
Comment (pierallard) [inline src/Akeneo/Channel/Bundle/EventListener/ChannelLocaleSubscriber.php]: +1

File: src/Akeneo/Channel/Bundle/EventListener/ChannelLocaleSubscriber.php [hunk 5]
```diff
@@ -106,40 +97,14 @@ public function updateChannel(GenericEvent $event)
...
-        }
+        $this->updateChannelInBackend($channel);
     }
 
     /**
      * Update the channel and if at least a locale is removed, it launches the completeness cleaning with a command.
      * @see https://akeneo.atlassian.net/browse/PIM-7155
      *
-     * @param ChannelInterface $channel
+     * @param \Akeneo\Channel\Component\Model\ChannelInterface $channel
```
Comment (pierallard) [inline src/Akeneo/Channel/Bundle/EventListener/ChannelLocaleSubscriber.php]: use

File: src/Akeneo/Channel/Component/spec/Normalizer/Versioning/ChannelNormalizerSpec.php [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
-use Pim\Component\Catalog\Model\ChannelInterface;
+use Akeneo\Channel\Component\Model\ChannelInterface;
 use Prophecy\Argument;
 use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 
 class ChannelNormalizerSpec extends ObjectBehavior
 {
     function let(
-        Standard\ChannelNormalizer $channelNormalizerStandard,
+        \Akeneo\Channel\Component\Normalizer\Standard\ChannelNormalizer $channelNormalizerStandard,
```
Comment (pierallard) [inline src/Akeneo/Channel/Component/spec/Normalizer/Versioning/ChannelNormalizerSpec.php]: :baby_chick:

File: src/Pim/Component/Catalog/Validator/AttributeValidatorHelper.php [hunk 7]
```diff
@@ -28,8 +28,8 @@ class AttributeValidatorHelper
     protected $scopeCodes;
 
     /**
-     * @param LocaleRepositoryInterface  $localeRepository
-     * @param ChannelRepositoryInterface $scopeRepository
+     * @param LocaleRepositoryInterface                                       $localeRepository
+     * @param \Akeneo\Channel\Component\Repository\ChannelRepositoryInterface $scopeRepository
```
Comment (pierallard) [inline src/Pim/Component/Catalog/Validator/AttributeValidatorHelper.php]: :dango:

File: UPGRADE.md (hunk 2 above)
Comment (arnolanglade) [reply UPGRADE.md]: yes

Comment (pierallard) [reply UPGRADE.md]: its name is "Localel" ?

**Pull request merged**